### PR TITLE
Added WithSecondaryNetwork func to deployment pkg

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/deployment/deployment.go [hunk 1]
```diff
@@ -169,6 +172,34 @@ func (builder *Builder) WithAdditionalContainerSpecs(specs []coreV1.Container) *
...
 
+// WithSecondaryNetwork applies Multus secondary network configuration on deployment definition.
+func (builder *Builder) WithSecondaryNetwork(network []*multus.NetworkSelectionElement) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying secondary network %v to deployment %s", network, builder.Definition.Name)
+
+	if builder.errorMsg != "" {
```
Comment (trewest) [inline pkg/deployment/deployment.go:183]: redundant with check in validate method

File: pkg/deployment/deployment.go [hunk 2]
```diff
@@ -169,6 +172,34 @@ func (builder *Builder) WithAdditionalContainerSpecs(specs []coreV1.Container) *
...
+		return builder
+	}
+
+	netAnnotation, err := json.Marshal(network)
```
Comment (trewest) [inline pkg/deployment/deployment.go:187]: should you check network length first?

Comment (kononovn) [reply pkg/deployment/deployment.go:187]: +1. Added

**Comment by kononovn** [comment]
@trewest PTAL

File: pkg/deployment/deployment.go [hunk 3]
```diff
@@ -169,6 +172,34 @@ func (builder *Builder) WithAdditionalContainerSpecs(specs []coreV1.Container) *
 	return builder
 }
 
+// WithSecondaryNetwork applies Multus secondary network configuration on deployment definition.
+func (builder *Builder) WithSecondaryNetwork(network []*multus.NetworkSelectionElement) *Builder {
```
Comment (evgenLevin) [inline pkg/deployment/deployment.go:176]: networkS []*multus.NetworkSelectionElement

**Review by trewest — DISMISSED** [review]
/lgtm

**Pull request merged**